nixos/modules/services/web-apps/stash.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.stash;
+in
+{
+  options = {
+    services.stash = {
+      enable = mkEnableOption "An organizer for your porn, written in Go.";
+
+      package = mkPackageOption pkgs "stash" { };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/stash";
+        description = "The state directory where stash stores its metadata, cache and generated files.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "stash";
+        description = "User account under which stash runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "stash";
+        description = "Group under which stash runs.";
+      };
+
+      ip = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
+        example = "::1";
+        description = "The ip that the server should bind to.";
+      };
```

Comment:
I suspect calling this ip is a bit misleading. It probably resolves strings such as `localhost` as well, so it should be called something like `host` or `listenAddress`.